website/docs/r/s3_bucket.html.markdown

```diff
@@ -12,438 +12,67 @@ Provides a S3 bucket resource.
-  }
-}
+### Object Lock Configuration
 
-resource "aws_s3_bucket" "source" {
-  provider = aws.central
-  bucket   = "tf-test-bucket-source-12345"
-  acl      = "private"
+-> **NOTE:** The `object_lock_configuration` attribute is deprecated.
+See [`aws_s3_bucket_object_lock_configuration`](s3_bucket_object_lock_configuration.html.markdown) for examples with object lock configurations on both new and existing buckets.
 
-  versioning {
-    enabled = true
-  }
+### Replication Configuration
 
-  replication_configuration {
-    role = aws_iam_role.replication.arn
-
-    rules {
-      id     = "foobar"
-      status = "Enabled"
-
-      filter {
-        tags = {}
-      }
-      destination {
-        bucket        = aws_s3_bucket.destination.arn
-        storage_class = "STANDARD"
-
-        replication_time {
-          status  = "Enabled"
-          minutes = 15
-        }
-
-        metrics {
-          status  = "Enabled"
-          minutes = 15
-        }
-      }
-    }
-  }
-}
-```
+-> **NOTE:** The `replication_configuration` attribute is deprecated.
+See [`aws_s3_bucket_replication_configuration`](s3_bucket_replication_configuration.html.markdown) for examples with replication configured.
 
 ### Enable SSE-KMS Server Side Encryption
 
--> **NOTE:** The parameter `server_side_encryption_configuration` is deprecated.
-Use the resource [`aws_s3_bucket_server_side_encryption_configuration`](s3_bucket_server_side_encryption_configuration.html) instead.
+-> **NOTE:** The `server_side_encryption_configuration` attribute is deprecated.
+See [`aws_s3_bucket_server_side_encryption_configuration`](s3_bucket_server_side_encryption_configuration.html.markdown) for examples with server side encryption configured.
 
-```terraform
-resource "aws_kms_key" "mykey" {
-  description             = "This key is used to encrypt bucket objects"
-  deletion_window_in_days = 10
-}
+### ACL Policy Grants
 
-resource "aws_s3_bucket" "mybucket" {
-  bucket = "mybucket"
-
-  server_side_encryption_configuration {
-    rule {
-      apply_server_side_encryption_by_default {
-        kms_master_key_id = aws_kms_key.mykey.arn
-        sse_algorithm     = "aws:kms"
-      }
-    }
-  }
-}
-```
-
-### Using ACL policy grants
-
--> **NOTE:** The parameters `acl` and `grant` are deprecated.
-Use the resource [`aws_s3_bucket_acl`](s3_bucket_acl.html.markdown) instead.
-
-```terraform
-data "aws_canonical_user_id" "current_user" {}
-
-resource "aws_s3_bucket" "bucket" {
-  bucket = "mybucket"
-
-  grant {
-    id          = data.aws_canonical_user_id.current_user.id
-    type        = "CanonicalUser"
-    permissions = ["FULL_CONTROL"]
-  }
-
-  grant {
-    type        = "Group"
-    permissions = ["READ_ACP", "WRITE"]
-    uri         = "http://acs.amazonaws.com/groups/s3/LogDelivery"
-  }
-}
-```
+-> **NOTE:** The `acl` and `grant` attributes are deprecated.
+See [`aws_s3_bucket_acl`](s3_bucket_acl.html.markdown) for examples with ACL grants.
 
 ## Argument Reference
```

Comment:
Not necessary now, but it would be good to move the (many) deprecated arguments below the (few) non-deprecated ones.